File: server/src/main/java/com/cafetamine/surveys/model/Category.java
package com.cafetamine.surveys.model;

import javax.persistence.*;
import javax.validation.constraints.NotEmpty;
import javax.validation.constraints.NotNull;

import java.util.Date;


@Entity
@Table(name = "categories")
public class Category extends AuditModel {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;
    @NotEmpty
    private String tag;
    @NotNull
    @OneToOne
    private User author;

    public Long getId() {
        return id;
    }

    public void setId(Long id) {
        this.id = id;
    }

    public String getTag() {
        return tag;
    }

    public void setTag(String tag) {
        this.tag = tag;
    }

    public User getAuthor() {
        return author;
    }

    public void setAuthor(User author) {
        this.author = author;
    }

}

File: client/src/app/model/category.ts
import { Deserializable } from "./deserializable";
import { User } from "./user";


export class Category implements Deserializable<Category> {
  id: number;
  tag: string;
  author: User;
  createdAt: Date;
  updatedAt: Date;

  public deserialize(object: any): Category {
    this.id = object.id;
    this.tag = object.tag;
    this.createdAt = object.createdAt;
    this.updatedAt = object.updatedAt;
    this.author = new User().deserialize(object.author);

    return this;
  }

}
